drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (c) 2023 Jan Henke <Jan.Henke@taujhe.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sparkfun_serlcd
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/auxdisplay.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(auxdisplay_serlcd, CONFIG_AUXDISPLAY_LOG_LEVEL);
+
+/*
+ * | in ASCII, used to begin a display command
+ */
+#define SERLCD_BEGIN_COMMAND 0x7C
+
+/*
+ * special command for the underlying display controller
+ */
+#define SERLCD_BEGIN_SPECIAL_COMMAND 0xFE
+
+union auxdisplay_display_state {
+	uint8_t byte;
+	struct {
+		uint8_t padding: 5;
+		bool power: 1;
+		bool cursor: 1;
+		bool blinking: 1;
+	};
```

Comment:
This should be marked as `__packed` and this should also have 2 layouts, one for big endian, one for little endian.

Reply:
Added packed. I disagree on the 2 layouts, as It's size is only 1 byte, for which the endianess does not make any difference.

Reply:
Actually for a bitfield, it does. Here is a modified hello_world sample:
```
#include <stdbool.h>
#include <stdint.h>
#include <stdio.h>

union auxdisplay_display_state {
        uint8_t byte;
        struct __attribute__((__packed__)) {
                uint8_t padding: 5;
                bool power: 1;
                bool cursor: 1;
                bool blinking: 1;
        };
};

union auxdisplay_display_state obj;

int main(void)
{
obj.power = 1;
obj.cursor = 1;
        printf("Hello World! %s\nValue: %x\n", CONFIG_BOARD, obj.byte);
        return 0;
}
```
Here is it running on a little endian ARM cortex:
```
*** Booting Zephyr OS build v3.5.0-rc1-38-gc6c97002b529 ***
Hello World! qemu_cortex_m3
Value: 60
```
And here is it running on a big endian sparc:
```
[93/94] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: leon3
*** Booting Zephyr OS build v3.5.0-rc1-38-gc6c97002b529 ***
Hello World! qemu_leon3
Value: 6
```

0x6 != 0x60